matscipy/calculators/manybody/newmb.py

```diff
@@ -641,3 +643,45 @@ def get_hessian(self, atoms, format='sparse', divide_by_masses=False):
         )
 
         return H
+
+class HessianPrecon(Precon):
+    def __init__(self, calc, c_stab=0.01, move_tol=0.1, P=None, old_positions=None):
+        self.calc = calc
+        self.P = P        
+        self.c_stab = c_stab
+        self.move_tol = move_tol
+        self.old_positions = old_positions        
+        
+    def make_precon(self, atoms):
+        if self.P is None or self.old_positions is None:
+            max_move = np.inf
+        else:
+            max_move = np.abs(atoms.positions - self.old_positions).max()
```

Comment:
ASE calculators have a `check_state` function that returns changes in the system, one could test `"positions" in self.calc.check_state(self.old_atoms)` or alternatively use the `compare_atoms` in `ase.calculators.calculator` to test for a single property.